lib/galaxy/managers/notification.py

```diff
@@ -291,7 +291,7 @@ def update_user_notification_preferences(
         """Updates the user's notification preferences with the requested changes."""
         notification_preferences = self.get_user_notification_preferences(user)
         notification_preferences.update(request.preferences)
-        user.preferences[NOTIFICATION_PREFERENCES_SECTION_NAME] = notification_preferences.json()
+        user.preferences[NOTIFICATION_PREFERENCES_SECTION_NAME] = notification_preferences.json()  # type:ignore[index]
         with transaction(self.sa_session):
```

Comment:
```suggestion
        user.preferences[NOTIFICATION_PREFERENCES_SECTION_NAME] = (
            notification_preferences.model_dump_json()
        )  # type:ignore[index]
```

Reply:
Same here... `.json()` to `.model_dump_json()` as `.json()` is deprecated in Pydantic 2

Reply:
Done!